### ISSUE TITLE
Add support for conversion CSV -> GPKG

### ISSUE THREAD

**Issue by schoeller** [anuga-community/anuga_core#39]
Dear all,

thanks for this project.
I see a need to convert from CSV -> GPKG file format starting out from `validation_tests\case_studies\merewether\houses\`. I have written a first script to compile from the buildings in CSV format to a GPKG-package. It serves my purposes and I would like to contribute as per my capacities.

Best regards

Sebastian

```python
"""
Sources:
https://stackoverflow.com/questions/9234560/find-all-csv-files-in-a-directory-using-python#12280052
https://gis.stackexchange.com/questions/301251/geopandas-import-csv-make-polygons-from-x-y-coordinates-based-on-id
https://gis.stackexchange.com/questions/294206/%d0%a1reating-polygon-from-coordinates-in-geopandas
https://gis.stackexchange.com/questions/266098/how-to-convert-a-geoseries-to-a-geodataframe-with-geopandas
"""

import os
from glob import glob
import pandas as pd
from shapely.geometry import Polygon
import geopandas as gpd

def get_csv_files(dir_path, ext):
    os.chdir(dir_path)
    return list(map(lambda x: os.path.join(dir_path, x), glob(f'*.{ext}')))

def get_mpolygons(file_list, crs):
    polygons = []
    for file in file_list:
        df = pd.read_csv(file, header=None)
        polygons.append(Polygon(zip(df[0], df[1])))
    gs = gpd.GeoSeries(polygons)
    print(gs)
    return gpd.GeoDataFrame(geometry=gpd.GeoSeries(gs, crs=crs))

if __name__ == "__main__":
    file_list = get_csv_files(".", "csv")
    mp = get_mpolygons(file_list, crs='epsg:32756')
    print(mp)
    mp.to_file(filename='buildings.gpkg', driver="GPKG")
```

**Comment by stoiver** [comment]
Hi @schoeller, thanks for the suggestion. I guess an alternative way of phrasing your request is to include geopandas objects as input to anuga. In a similar vain, I have been looking including `xarray` objects as input to `Rate_operators`

**Comment by schoeller** [comment]
Hi @stoiver and thanks for the feedback. 
Certainly one could include a [DataFrame](https://geopandas.org/en/stable/docs/user_guide/data_structures.html#geodataframe) as an input object, as it might ease data handling.
I would think that a modeller is using GIS data as per their contractual or working conditions (custom made to their own purposes). A dataobject introduced into the ANUGA system should in my opinion be defined as per minimum standard required for the model itself. It cannot match all modellers demands. In consequence it might well be possible, that the modeller still is required to translate working conditions to an ANUGA required input definition.
In conclusion I would think that a conversion from all vector-based input information to one GPKG-based database could be of interest in order to gain direct access through GIS and thus open data handling to a broader user base as it eases the workflow. The (reasonable) barrier to use Python for carrying out ANUGA modelling tasks persists though.

**Comment by stoiver** [comment]
Done in #370 (merged to develop): `polygon_csv_files2gpkg` / `building_csv2gpkg` write the two ANUGA CSV polygon conventions to a GeoPackage layer, and `polygons2gpkg` writes any list of ANUGA polygons with attributes and a CRS. Thanks for the scripts that pointed the way.
